src/utils/Utils.ts

```diff
@@ -79,4 +93,30 @@ export function getPotentialRoman(str: string): number {
             return parseInt(str);
         }
     })()
+}
+
+export function calculateSeasonAndEpisode(programme: IPlayerProgramMetadata) : [ type: VideoType, episode?: number, episodeTitle?: string, series?: number ] {
```

Comment:
Refactored this method a _lot_ to get it to be relatively readable but still pass the tests.